pysteps/blending/steps.py

```diff
@@ -1538,19 +1545,35 @@ def worker(j):
                             # Set to min value outside of mask
                             R_f_new[~MASK_prec_] = R_cmin
 
+                        # If probmatching_method is not None, resample the distribution from
+                        # both the extrapolation cascade and the model (NWP) cascade and use
+                        # that for the probability matching
+                        if probmatching_method is not None and resample_distribution:
+                            R_pm_resampled = probmatching.resample_distributions(
+                                a=R_pm_ep[t_index],
+                                b=precip_models_pm_temp[j],
+                                weight=weights_pm_normalized[
+                                    0
+                                ],  # Use the weight from the extrapolation cascade here
```

Comment:
should we specify somewhere that the weights are based on level 2?

Reply:
Good call, I will add it!